Vindicta.Altis/MessageReceiver/MessageReceiver.sqf

```diff
@@ -32,8 +32,10 @@ MsgRcvr_fnc_setMsgDone = {
 	pr _rqArrayElement = g_rqArray select _msgID; // g_rqArray was defined in messageReceiver.sqf
 	// Make sure the proper receiver marks this message
 	if ((_rqArrayElement select 2) == _dest) then {
-		_rqArrayElement set [0, 1]; // Set the flag that the message has been processed
-		_rqArrayElement set [1, _result];
+		CRITICAL_SECTION {
+			_rqArrayElement set [1, _result];	// Set result first
```

Comment:
So it causes a race condition? Fascinating. How does it happen?

Reply:
The other thread reads the flag to determine if the message is handled, then returns the value if the flag is set. But the flag is set before the value so if the scheduler decides to only set the flag, then switch to the other thread it will not set the result in time.

Reply:
You are obviously write. Should first write the result, then set the flag. Having a critical section is even safer. Nice that we have it fixed finally! :thumbsup: